nodes/modules/workers/workers.tf

```diff
@@ -1,10 +1,10 @@
 resource "aws_autoscaling_group" "nodes" {
-  count            = var.nodes_enabled ? length(var.nodes_subnet_group) : 0
+  count            = var.nodes_enabled ? 1 : 0
   desired_capacity = var.desired_nodes
   max_size         = var.max_nodes
   min_size         = var.min_nodes
-  name             = "${var.node_group_name}-${aws_launch_template.node[count.index].id}-nodes-${count.index}"
-  vpc_zone_identifier = [element(var.nodes_subnet_group, count.index)]
+  name             = "${var.node_group_name}-${aws_launch_template.node[0].id}-nodes"
+  vpc_zone_identifier = var.nodes_subnet_group
```

Comment:
`nodes_subnet_group` is a list variable so not sure if this works or requires element index -
`var.nodes_subnet_group[0]`

Reply:
This object accepts lists, so it should be ok